### ISSUE TITLE
Quicker fade-outs; fade preference

### ISSUE THREAD

**Issue by SeanMSmith** [JohnCoates/Aerial#534]
I'm getting into the realm of picking very small nits here, but I begrudge every moment of stunning 4K detail of our planet lost during the relatively slow fade-outs. Would it be possible to speed up the fade-outs, so that they match the speed of the fade-ins, and perhaps to add a preference to disable fades altogether?

**Comment by glouel** [comment]
Fade in/out should be 2 seconds IIRC. There's one video (one of the Space China ?) that seems to fade infinitely but it's actually the video that ends like that with super low visibility for many seconds, it puzzled me once too. Are you sure that's not what you experienced ?

Disable and speed if so, why not though 2 secs I though would be a good default, we can find a better default too maybe if that's really too slow.

**Comment by SeanMSmith** [comment]
I'm seeing two-second fade-outs (which are too slow for my taste) and what appear to be much quicker fade-ins (which I don't mind).

**Comment by glouel** [comment]
Ok that makes sense, I was looking at the code and it seemed like fade outs should be 2s. You're seeing quicker fade ins because they start before the video actually shows up (which can take half a sec/a sec). I'll try to make it stop doing that and add something to switch to shorter fades. 

1s maybe ?

**Comment by glouel** [comment]
When you got a sec, check 
https://github.com/glouel/Aerial/releases/download/1.4.1test2/Aerial.saver.zip

My heart says 0.5 default 😱

I only adjusted the video fade, I'll align the text fades to the same value I guess.

**Comment by SeanMSmith** [comment]
Fades are _much_ faster in 1.4.1test2; in fact, I seriously doubt if the total time from the beginning of the fade-out to the completion of the fade-in is as much as 0.5s now.

I don't mind the much faster fades and prefer them to those in 1.4, but some might find them abrupt.

**Comment by glouel** [comment]
It may look shorter because it's a linear fade, while you may be used to better ease in/ease out fades, but the time is correct. 

Fade ins were especially not correct in default 1.4 and cut by your video initialization time. I'll look into ease in/out to get a better result and see where that goes.

**Comment by glouel** [comment]
So I added cubic ease which makes them feel more natural, but the cubic function may make them feel even shorter. I went with 1s default on that build for new users, can you let me know what you think when you have a sec? 

https://github.com/glouel/Aerial/releases/download/1.4.1test3/Aerial.saver.zip

PS : This build also should have extended descriptions on old videos. Try London, SF or NY !

**Comment by SeanMSmith** [comment]
You've been busy!

This is interesting. In 1.4.1 beta 3, I am now seeing 0.5 second fades when set to 0.5 seconds, 1 second fades when set to 1 second, and 2 second fades when set to 2 seconds. (I never tested 1.4.1 beta 1 or 2, or test 4.)

In 1.4.1 test 2, as I reported previously, fades were _much_ faster than 0.5 seconds, and the same was also true with 1.4.1 test 3. (In fact, I could discern no difference between test 2 and test 3.)

(In every case, I was careful to close and re-open System Preferences after installing the new version of Aerial.)

In other words, something has changed besides the user interface.

At any rate, all seems to work correctly now, and I'm quite happy with the 0.5 second video fades.

Thank you!

**Comment by glouel** [comment]
A bit ;) 

I did change the fade to add a cubic ease, they look more natural now (probably the difference you feel). 

I'll keep those 4 values, I think they give a good balance. All texts (including the new clock) use the same fade time too.

**Comment by glouel** [comment]
I have a hard time keeping up, I made too many builds ! As long as they are correct now... ;) (and see my answer there)

**Comment by glouel** [comment]
Closing as fixed with the release of 1.4.1
